test/optimizers/test_optimizers.py

```diff
@@ -141,6 +146,72 @@ def test_sr3_bad_parameters(optimizer, params):
         optimizer(**params)
 
 
+@pytest.mark.parametrize(
+    "params",
+    [
+        dict(eta=-1),
+        dict(tol=0),
+        dict(tol_m=0),
+        dict(eps_solver=0),
+        dict(alpha_m=-1),
+        dict(alpha_A=-1),
+        dict(gamma=1),
+        dict(evolve_w=False, relax_optim=False),
+        dict(thresholder="l0"),
+        dict(threshold=-1),
+        dict(max_iter=0),
+        dict(eta=10, alpha_m=20),
+        dict(eta=10, alpha_A=20),
+    ],
+)
+def test_trapping_bad_parameters(params):
+    with pytest.raises(ValueError):
+        TrappingSR3(**params)
+
+
+@pytest.mark.parametrize(
+    "params",
+    [dict(PL=np.random.rand(3, 3, 3, 9)), dict(PQ=np.random.rand(3, 3, 3, 3, 9))],
+)
+def test_trapping_bad_tensors(params):
+    x = np.random.standard_normal((10, 9))
+    x_dot = np.random.standard_normal((10, 3))
+    with pytest.raises(ValueError):
+        model = TrappingSR3(**params)
+        model.fit(x, x_dot)
+
+
+@pytest.mark.parametrize(
+    "params",
+    [dict(PL=np.ones((3, 3, 3, 9)), PQ=np.ones((3, 3, 3, 3, 9)))],
+)
+def test_trapping_quadratic_library(params):
+    x = np.random.standard_normal((10, 3))
+    library_functions = [
+        lambda x: x,
+        lambda x, y: x * y,
+        lambda x: x ** 2,
+        lambda x, y, z: x * y * z,
+        lambda x, y: x ** 2 * y,
+        lambda x: x ** 3,
+    ]
+    library_function_names = [
+        lambda x: x,
+        lambda x, y: x + y,
+        lambda x: x + x,
+        lambda x, y, z: x + y + z,
+        lambda x, y: x + x + y,
+        lambda x: x + x + x,
+    ]
+    sindy_library = CustomLibrary(
+        library_functions=library_functions, function_names=library_function_names
+    )
+    with pytest.raises(ValueError):
```

Comment:
Why should a `ValueError` be raised here? Shouldn't `TrappingSR3` work with a library like this one?

Reply:
I can add some more documentation here. The answer is yes, Trapping SR3 can work with libraries other than quadratic. What is can't do is work with non-quadratic libraries AND use a defined PL and PQ tensor, since this implies that the trapping theorem is being used (and hence the library must be no more than quadratic in polynomials).

Reply:
Ah I see. In that case do you think you could add a test where `PL` and `PQ` are used correctly?

Reply:
sounds good, will do